client-libraries/python/auxon-sdk/src/ingest.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{types, SdkError};
+use auxon_sdk::{
+    api::{AttrVal, TimelineId},
+    ingest_client::IngestStatus,
+};
+use pyo3::prelude::*;
+
+#[pyclass]
+pub struct IngestClient {
+    rt: tokio::runtime::Runtime,
+    client: auxon_sdk::plugin_utils::ingest::Client,
+}
+
+impl IngestClient {
+    pub fn new(
+        rt: tokio::runtime::Runtime,
+        client: auxon_sdk::plugin_utils::ingest::Client,
+    ) -> Self {
+        Self { rt, client }
+    }
+}
+
+#[pymethods]
+impl IngestClient {
+    pub fn disable_auto_timestamp(&mut self) {
+        self.client.disable_auto_timestamp();
+    }
+
+    pub fn switch_timeline(&mut self, id: TimelineId) -> Result<(), SdkError> {
+        Ok(self.rt.block_on(self.client.switch_timeline(id))?)
+    }
+
+    pub fn send_timeline_attrs(
+        &mut self,
+        name: &str,
+        timeline_attrs: &Bound<pyo3::types::PyDict>,
+    ) -> Result<(), PyErr> {
+        let attrs = py_dict_to_attr_vec(timeline_attrs)?;
+        self.rt
+            .block_on(
+                self.client
+                    .send_timeline_attrs(name, attrs.iter().map(|(k, v)| (k.as_str(), v.clone()))),
+            )
+            .map_err(SdkError::from)?;
```

Comment:
I'm surprised we can't do the usual `impl From` and use implicit error type conversion in place of these `map_err`'s?

Reply:
I tihnk we probably can in this case.